### ISSUE TITLE
eslint@8.x in conjunction with @typescript-eslint/parser@5.x breaks eslint camelcase rule

### ISSUE THREAD

**Issue by MhMadHamster** [typescript-eslint/typescript-eslint#4194]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [ ] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/main/docs/getting-started/linting/TROUBLESHOOTING.md) and my problem is not listed.

**Repro**

```JSON
{
  "plugins": [
    "@typescript-eslint"
  ],
  "overrides": [
    {
      "files": "*.{ts,tsx}",
      "parser": "@typescript-eslint/parser",
    }
  ],
  "rules": {
    "camelcase": "error",
  },
}
```

```TS
// your repro code case
interface TestInterface {
  snake_case: string;
}
```

**Expected Result**
Error about `snake_case` not being in camelcase

**Actual Result**
No error

**Additional Info**
I am not sure if problem lies within `eslint` or `@typescript-eslint/parser`. `camelcase` rule still works well in js files in latest version of `eslint`, but breaks in `*.ts` files, so I am assuming that's something to do with parsing.

**Versions**

| package                     | version |
| --------------------------- | ------- |
| `@typescript-eslint/parser` | `5.4.0` |
| `TypeScript`                | `4.5.2` |
| `ESLint`                    | `8.2.0` |
| `node`                      | `16.13.0` |

**Comment by bradzacher** [comment]
https://typescript-eslint.io/rules/camelcase

We do not support camelcase on TS constructs and have not since v3 (over a year ago).

**Comment by MhMadHamster** [comment]
Ah, ok, was just confused because it was working using `@typescript-eslint/parser@5.x` and `eslint@7.x` and didn't saw anything in changelog, but apparently its not supported since v.3 😅
